src/private/MDMPerformerGroup.m

```diff
@@ -76,30 +78,64 @@ - (void)terminateIsActiveToken:(id<MDMIsActiveTokenable>)token
 #pragma mark - Private
 
 - (void)addPlan:(id<MDMPlan>)plan trace:(MDMTrace *)trace log:(MDMTransactionLog *)log {
-  [trace.committedPlans addObject:plan];
-  
+  // all named addPlan plans must first be removed before being added
+  if ([self isNamedTransactionLog:log] && !log.isRemoval) {
+    [self addPlan:plan trace:trace log:[[MDMTransactionLog alloc] initWithPlans:@[plan] target:log.target name:log.name removal:TRUE]];
+  }
+  // see if we can get a performer
+  id<MDMPerforming> performer = [self performerForPlan:plan trace:trace log:log];
+  if (performer != nil) {
+    if (plan != nil) {
+      [trace.committedPlans addObject:plan];
+    } else {
+      // this is the case whereby we are calling removePlan:named, but don't have a MDMPlan
```

Comment:
I see that `Tracer` PR (http://codereview.cc/D1682) landed with the `didAddPlan(_:named:to:)` method. As this PR includes logic for removing named plans, would you suggest a similar `didRemovePlanNamed(_:from:)` `Tracer` optional method?